### DISCUSSION TITLE
How do I train LoFTR on a custom dataset?

### DISCUSSION THREAD

**Discussion by adityamwagh** [zju3dv/LoFTR#178] [Q&A]
The training readme only shows information about2 datasets. How do I train on a custom dataset?

**Comment by HUSTNO1WXY** [comment]
I wonder that how you solved this issue.

**Reply by adityamwagh** [reply]
Someone has a written a LoFTR training notebook using [Kornia](https://github.com/kornia/kornia) for the [Image Matching Challenge](https://www.kaggle.com/competitions/image-matching-challenge-2022) on Kaggle

You can check [this](https://github.com/adityamwagh/hpml-project) repository.

**Comment by zyxzyx45** [comment]
请问一下解决了么

**Reply by adityamwagh** [reply]
Yes.

**Comment by ZahraShahlaie** [comment]
![image](https://github.com/user-attachments/assets/d3b4889d-5859-4075-977f-fa4b17f2efe5)

تصویر بالا را یافتم هم به [این کد](https://www.kaggle.com/code/ammarali32/training-loftr) توجه میکنم [هم به این ریپازیتوری ](https://github.com/adityamwagh/pose-estimation-loftr/tree/main)و سعی میکنم خودم بخشی را پیاده سازی کنم با کد های اصلی مقایسع کنم ساختار دیتا را درک کنم و ...

**Reply by adityamwagh** [reply]
LoFTR is kinda old at this point. You should check out these models instead.
- [Efficient LoFTR](https://github.com/zju3dv/efficientloftr)
- [DKM: Dense Kernelized Feature Matching for Geometry Estimation](https://github.com/Parskatt/dkm)
- [MatchAnything](https://github.com/zju3dv/MatchAnything)

Zero-shot performance using MatchAnything is quite good, so you might not need to fine-tune.
